loadbalancer/dnssrv/publisher_internal_test.go

```diff
@@ -0,0 +1,170 @@
+package dnssrv
+
+import (
+	"errors"
+	"net"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/loadbalancer"
+	"github.com/go-kit/kit/log"
+)
+
+func TestPublisher(t *testing.T) {
+	var (
+		target = "my-target"
+		port   = uint16(1234)
+		addr   = &net.SRV{Target: target, Port: port}
+		addrs  = []*net.SRV{addr}
+		name   = "my-name"
+		ttl    = time.Second
+		logger = log.NewNopLogger()
+		e      = func(context.Context, interface{}) (interface{}, error) { return struct{}{}, nil }
+	)
+
+	oldLookup := lookupSRV
+	defer func() { lookupSRV = oldLookup }()
+	lookupSRV = mockLookupSRV(addrs, nil, nil)
+
+	factory := func(instance string) (endpoint.Endpoint, error) {
+		if want, have := addr2instance(addr), instance; want != have {
+			t.Errorf("want %q, have %q", want, have)
+		}
+		return e, nil
+	}
+
+	p, err := NewPublisher(name, ttl, factory, logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer p.Stop()
+
+	if _, err := p.Endpoints(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestBadLookup(t *testing.T) {
+	oldLookup := lookupSRV
+	defer func() { lookupSRV = oldLookup }()
+	lookupSRV = mockLookupSRV([]*net.SRV{}, errors.New("kaboom"), nil)
+
+	var (
+		name    = "some-name"
+		ttl     = time.Second
+		factory = func(string) (endpoint.Endpoint, error) { return nil, errors.New("unreachable") }
```

Comment:
See: http://play.golang.org/p/MGk22izFHt

Just an idea to don't repeat too much code